core/templates/dev/head/pages/collection_editor/CollectionEditorNavbarDirective.js

```diff
@@ -68,11 +78,12 @@ oppia.directive('collectionEditorNavbar', [function() {
           // action since it is not reversible.
           CollectionRightsBackendApiService.setCollectionPublic(
             $scope.collectionId, $scope.collection.getVersion()).then(
-            function() {
-              // TODO(bhenning): There should be a scope-level rights object
-              // used, instead. The rights object should be loaded with the
-              // collection.
-              $scope.isPrivate = false;
+            function(data) {
+              $scope.isPrivate = collectionRightsObject.setPublic(data);
+              if ($scope.isPrivate) {
+                alertsService.addWarning(
```

Comment:
Sorry -- can you tell me again why this check can't go into CollectionRightsBackendApiService.setCollectionPublic()? I mean, you could do the check on success, and then delegate to the passed-in callback...

Then, the error-checking is transparent to the controller, and all the success callback here would need to do is to update the rights object accordingly.